drivers/sensor/rpi_pico_temp/rpi_pico_temp.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2023 TOKITA Hiroshi
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/logging/log.h>
+
+#include <hardware/adc.h>
+
+LOG_MODULE_REGISTER(rpi_pico_temp, CONFIG_SENSOR_LOG_LEVEL);
+
+#define DT_DRV_COMPAT raspberrypi_pico_temp
+
+struct rpi_pico_temp_config {
+	const struct device *adc;
+	const struct adc_channel_cfg ch_cfg;
+};
+
+struct rpi_pico_temp_data {
+	struct adc_sequence adc_seq;
+	int16_t sample;
+};
+
+static int rpi_pico_temp_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct rpi_pico_temp_config *cfg = dev->config;
+	struct rpi_pico_temp_data *data = dev->data;
+	int rc;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_DIE_TEMP) {
+		return -ENOTSUP;
+	}
+
+	rc = adc_channel_setup(cfg->adc, &cfg->ch_cfg);
+	if (rc) {
+		LOG_DBG("Setup AIN%u got %d", cfg->ch_cfg.channel_id, rc);
+		return rc;
+	}
+
+	rc = adc_read(cfg->adc, &data->adc_seq);
+
+	return rc;
+}
+
+static int rpi_pico_temp_channel_get(const struct device *dev, enum sensor_channel chan,
+				     struct sensor_value *val)
+{
+	const struct rpi_pico_temp_config *cfg = dev->config;
+	struct rpi_pico_temp_data *data = dev->data;
+	int32_t mv = 0;
+	int32_t work;
+	int rc;
+
+	if (chan != SENSOR_CHAN_DIE_TEMP) {
+		return -ENOTSUP;
+	}
+
+	mv = data->sample;
+	rc = adc_raw_to_millivolts(adc_ref_internal(cfg->adc), cfg->ch_cfg.gain,
+				   data->adc_seq.resolution, &mv);
+	if (rc) {
+		LOG_DBG("adc_raw_to_millivolts() failed %d", rc);
+		return rc;
+	}
+
+	/*
+	 * Calculate CPU temperature from voltage by the equation:
+	 * T = 27 - (ADC_Voltage - 0.706)/0.001721
+	 */
+	work = ((27 * 1721) - (mv * 1000 - 706000));
```

Comment:
As the datasheet states:

> As the Vbe and the Vbe slope can vary over the temperature range, and from device to device, some user calibration may be required if accurate measurements are required.

So, I think these constants should come from DTS, with the values you've used as the defaults.

Reply:
Added defines `vbe` and `vbe-slope` to the node.